File: Julia/Root Estimation/fixedpointiteration.jl
function fixedpointiteration(g::Function, xi::Number, ea::Number, maxIter::Int64)
  # Estiamte roots for an equation using the fixed-point iteration method
  for i = 1:maxIter
    xiold = xi;
    xi = g(xi);

    if (abs(xi - xiold) / xi) < ea
      break;
    end;
  end
  
  return xi;
end

File: MATLAB/Root Estimation/fixedpointiteration.m
function [xi] = fixedpointiteration(g, xi, ea, maxIter)
%fixedpointiteration Estimate roots using the fixed point iteration method
%   xr = fixedpointiteration(g,xi,ea,maxIter) estimates the root xr of a
%   function using the rearrangement g of the function in terms of x. Using
%   xi as an initial estimate, it iterates on the root until the relative
%   error falls below ea or maxIter iterations occur.   
  for ii = 1:maxIter
    xiold = xi;
    xi = g(xi);
    
    if (abs(xi - xiold) / xi) < ea && ii > 1
      break;
    end
  end
end
